src/AWS.Deploy.CLI/Commands/TypeHints/InstanceCPUCoresCommand.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AWS.Deploy.Common;
+using AWS.Deploy.Common.Recipes;
+using AWS.Deploy.Common.TypeHintData;
+using AWS.Deploy.Orchestration.Data;
+
+namespace AWS.Deploy.CLI.Commands.TypeHints
+{
+    public class InstanceCPUCoresCommand : ITypeHintCommand
+    {
+        private readonly IAWSResourceQueryer _awsResourceQueryer;
+        private readonly IConsoleUtilities _consoleUtilities;
+        private const string INSTANCE_FREE_TIER_ELIGIBLE = "InstanceFreeTierEligible";
+        private const string INSTANCE_ARCHITECTURE = "InstanceArchitecture";
+
+        public InstanceCPUCoresCommand(IAWSResourceQueryer awsResourceQueryer, IConsoleUtilities consoleUtilities)
+        {
+            _awsResourceQueryer = awsResourceQueryer;
+            _consoleUtilities = consoleUtilities;
+        }
+
+        private async Task<List<int>?> GetData(Recommendation recommendation, OptionSettingItem optionSetting)
+        {
+            var parentSetting = recommendation.GetOptionSetting(optionSetting.ParentSettingId);
+
+            var freeTierEligible = parentSetting.ChildOptionSettings.FirstOrDefault(x => x.Id.Equals(INSTANCE_FREE_TIER_ELIGIBLE));
```

Comment:
We shouldn't have string value type hint that is relying on other sibling settings. What if a recipe wants to use the instance type typehint but it doesn't define an `InstanceFreeTierEligible` or `InstanceArchitecture`

Reply:
I removed the extra settings